src/pages/workspaces/workspace/WorkspaceMenu.js

```diff
@@ -0,0 +1,90 @@
+import { Fragment } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import { Clickable } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import { makeMenuIcon, MenuButton, MenuTrigger } from 'src/components/PopupTrigger'
+import { useWorkspaceDetails } from 'src/components/workspace-utils'
+import * as Utils from 'src/libs/utils'
+
+
+// In `workspaceInfo`, specify either `name and namespace` to fetch the Workspace details,
+// or `canShare, isAzureWorkspace, isLocked, and isOwner` to use previously fetched details.
+const WorkspaceMenu = ({
+  iconSize, popupLocation,
+  callbacks: { onClone, onShare, onLock, onDelete },
+  workspaceInfo: { name, namespace, canShare, isAzureWorkspace, isLocked, isOwner }
+}) => {
+  const navIconProps = {
```

Comment:
Note that these properties weren't used on the Workspaces list before, but I think the slightly different hover state is nice and it is good to be consistent.